modules/videoio/src/cap_aravis.cpp

```diff
@@ -102,7 +102,7 @@ class CvCaptureCAM_Aravis : public CvCapture
         close();
     }
 
-    virtual bool open(int);
+    virtual bool open(int, const cv::VideoCaptureParameters&);
```

Comment:
Lets avoid modification of all backends implementation.
Use virtual function of base class instead.

Modify 1-2 backend which are interested and can be used as sample.

Reply:
add the open(..., const cv :: VideoCaptureParameters &) function to the base class and implement it only in the required backends (example: gstreamer, msmf)?

Reply:
yes.

Also check similar patch for `VideoWriter` from here: #16766

Reply:
Still don't get it. Do you mean that you need to take all the open () to the top in the base class? Can you tell us more about your idea. When implementing the parameters of the video writer, the implementation of the open () function and the extraction of parameters remains at the backend level.

Reply:
Add common base class method:

```
bool CvCapture::open(int idx, const cv::VideoCaptureParameters& params)
{
    bool res = open(idx);
    if (res && !params.empty())
    {
        res &= setParameters(params);
        if (!res)
            close();
    }
    return res;
}
```

---

Derived classes which doesn't care about new API doesn't redefine this method.

Only FFmpeg / GStreamer / MSMF backends may have such method as an example (but do not bring large functionality there - it is not in scope of this PR). If you don't understand then don't add this overload except base classes.